xbmc/platform/darwin/tvos/TopShelf/ServiceProvider.m

```diff
@@ -102,20 +100,27 @@ - (TVTopShelfContentStyle)topShelfStyle
     return contentItems;
   };
 
-  if ([movieArray count] > 0)
-  {
-    __auto_type itemMovie = [[TVContentItem alloc] initWithContentIdentifier:wrapperIdentifier];
-    itemMovie.title = [(sharedDict ?: shared) valueForKey:@"moviesTitle"];
-    itemMovie.topShelfItems = contentItemsFrom(movieArray);
-    [topShelfItems addObject:itemMovie];
-  }
+  // Add each cateogyr to TopShelf
+  __auto_type topShelfItems = [[NSMutableArray alloc] init];
 
-  if ([tvArray count] > 0)
+  for (id categoryKey in sharedDict)
   {
-    __auto_type itemTv = [[TVContentItem alloc] initWithContentIdentifier:wrapperIdentifier];
-    itemTv.title = [(sharedDict ?: shared) valueForKey:@"tvTitle"];
-    itemTv.topShelfItems = contentItemsFrom(tvArray);
-    [topShelfItems addObject:itemTv];
+    if ([[sharedDict objectForKey:categoryKey] isKindOfClass:[NSDictionary class]])
```

Comment:
1. if not then continue
2. shorthand accessor here and in the condition body below